### PR TITLE
add sidesheet header

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description
Added the sidesheet header from Castberg, some functionality like the menu in the header is disabled for now. I think we need to revise how we want the user to configure this behaviour

Completes: AB#FILL_IN_YOUR_ISSUE_ID

## Type of change

Remove options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [x] My code is easy to read.
-   [x] Documentation and comments is added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: packages/workspace-fusion/src/lib/classes/fusionWorkspaceBuilder.tsx [hunk 1]
```diff
@@ -9,45 +9,77 @@ import {
...
+	color: string;
+}
+
+export interface FusionContext {
+	ui: FusionUI;
+}
+
+export class FusionWorkspaceBuilder<TData, TError> {
+	/** The name of your workspace/application */
+	name: string;
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/classes/fusionWorkspaceBuilder.tsx:23]: might rename "name" to appKey if this is supposed to be e.g. "handover", "query", "swcr"

File: packages/workspace-fusion/src/lib/components/Sidesheet/styledHeader.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import styled from 'styled-components';
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/components/Sidesheet/styledHeader.ts:1]: rename styling files to .styles.ts

File: packages/workspace-fusion/src/lib/components/Sidesheet/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+		<StyledHeader>
+			<StyledLeftHeader>
+				<StyledColourTab appColor={color}>
+					<Icon name="chevron_right" size={24} color={'white'} />
+				</StyledColourTab>
+				<StyledTitle>{title}</StyledTitle>
+			</StyledLeftHeader>
+
+			<StyledRightHeader>
+				{/* {menuItems.length > 0 && <IconMenu placement="bottom" items={menuItems} />} */}
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/components/Sidesheet/Header.tsx:22]: remove comment code

File: packages/workspace-fusion/src/lib/components/Sidesheet/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	title: string;
+	close: () => void;
+}
+
+export function SidesheetHeader({ close, color, title }: SidesheetHeaderProps) {
+	return (
+		<StyledHeader>
+			<StyledLeftHeader>
+				<StyledColourTab appColor={color}>
+					<Icon name="chevron_right" size={24} color={'white'} />
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/components/Sidesheet/Header.tsx:16]: color should be from tokens

File: packages/workspace-fusion/src/lib/classes/fusionWorkspaceBuilder.tsx (hunk 1 above)
Comment (Gustav-Eikaas) [reply packages/workspace-fusion/src/lib/classes/fusionWorkspaceBuilder.tsx:23]: Agreed

File: packages/workspace-fusion/src/lib/components/Sidesheet/SidesheetWrapper.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	 */
+
+	if (!clickEvent?.item) {
+		return null;
+	}
+
+	return (
+		<div>
+			<SidesheetHeader
+				color={controller.getContext()?.ui.color ?? 'white'}
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/components/Sidesheet/SidesheetWrapper.tsx:27]: color from tokens

File: packages/workspace-fusion/src/lib/types/fusionController.ts [hunk 6]
```diff
@@ -1,14 +1,15 @@
...
  * Workspace controller for fusion with some predefined types
  */
-export type FusionWorkspaceController<TData, TError, TContext> = WorkspaceController<
+export type FusionWorkspaceController<TData, TError> = WorkspaceController<
 	TData,
 	WorkspaceControllers<TData>,
 	WorkspaceOnClick<TData>,
 	TError,
-	TContext
+	FusionContext
```
Comment (jaysencpp) [inline packages/workspace-fusion/src/lib/types/fusionController.ts:14]: FusionContext refers to the actual "fusionContext" in Fusion?

Comment (Gustav-Eikaas) [reply packages/workspace-fusion/src/lib/types/fusionController.ts:14]: I agree this seems confusing, Its a context interface to add custom properties to the workspace controller. Should probably call it something like WorkspaceContext, Will store our custom stuff like fusion context id, name, color, etc etc..

Comment (jaysencpp) [reply packages/workspace-fusion/src/lib/types/fusionController.ts:14]: Should be renamed to TWorkspaceContext to avoid con-fusion 🤠

Comment (Gustav-Eikaas) [reply packages/workspace-fusion/src/lib/types/fusionController.ts]: But its not a generic

**Review by jaysencpp — APPROVED** [review]

**Pull request merged**